src/primitives/transaction.cpp

```diff
@@ -57,8 +57,10 @@ JSDescription JSDescription::Randomized(
     // Randomize the order of the inputs and outputs
     inputMap = {0, 1};
     outputMap = {0, 1};
-    MappedShuffle(inputs.begin(), inputMap.begin(), ZC_NUM_JS_INPUTS, gen);
-    MappedShuffle(outputs.begin(), outputMap.begin(), ZC_NUM_JS_OUTPUTS, gen);
+    if (gen) {
```

Comment:
I would prefer to pass in a no-op function rather than an empty one, ie. one that always returns `N_MAX-1` (which causes a shuffle that results in itself). See `GenMax()` in `src/gtest/utils.cpp` and its effect in `src/gtest/test_random.cpp`.